### ISSUE TITLE
Error message about group matcher when using 'try!' button in Regex Extractor builder.

### ISSUE THREAD

**Issue by bitginsu** [Graylog2/graylog2-server#3470]
## Expected Behavior
I'd expect the 'Try!' button to test my entered regex against the loaded log message above and tell me what it matched. 

## Current Behavior
I enter some regex on the extractor form for my Syslog UDP Input  (no matter how simple) and is shows a yellow bar saying: 'Regular expression does not contain any matcher group to extract.' 
I have not found any documentation on what a 'group matcher' much less on how to fix it. Seems to have something to do with Drools? 

## Possible Solution
Fix the 'group matcher'?

## Steps to Reproduce (for bugs)
1. Go to a log entry (maybe for a Syslog UDP source?) and expand it.
2. Choose the 'Create Extractor For -> Regex' button, I was using the 'message' or 'full_message' fields. 
3. On the extractor form, enter some regex that matches the example log entry.
4. Hit the 'Try!' button. 
5. A yellow 'Attention' bar will pop up saying 'Regular expression does not contain any matcher group to extract.' 

## Context
I'm trying to parse some Cisco ASA logs into more fields than either of the bundles in the Marketplace will do. (They give an error when attempting to import them in any case. marksie1988 has working version on his blog but not github: http://spottedhyena.co.uk/graylog2-cisco-asa-cisco-catalyst/#comment-233 )

## Your Environment
Using Graylog OVA 2.1.2 as a test box on VMWare to see if we can use Graylog in production. 

* Graylog Version: Graylog OVA 2.1.2 
* Elasticsearch Version:
* MongoDB Version:
* Operating System: Windows 10
![graylog-error](https://cloud.githubusercontent.com/assets/972938/22719710/a1002fe4-ed5b-11e6-9a24-d96aa0c7a57a.PNG)
![graylog-error2](https://cloud.githubusercontent.com/assets/972938/22719748/df236c6e-ed5b-11e6-9797-f8662fe5d15a.PNG)

* Browser version: Opera 42, Chrome 56

**Comment by joschi** [comment]
@bitginsu The problem is pretty much described in the error message: You don't have any matcher group to extract, i. e. while it's a regular expression, you didn't specify what to extract from the string.

For example this would work (taken from your 2nd screenshot) and would extract the first 3 characters: `^(\w{3})`.

We are using GitHub issues for tracking bugs in Graylog itself, but this doesn't look like one. Please post this issue to our [public mailing list](https://groups.google.com/forum/#!forum/graylog2) or join the [#graylog channel on freenode IRC](https://webchat.freenode.net/?channels=%23graylog).

Thank you!

**Comment by bitginsu** [comment]
Thanks very much for the quick answer,sorry if this is wrong venue. 
I went on freenode yesterday and asked this question, there, was no one home in #graylog. Also joined Google group. Will join mailing list now. 
 I'm not the most experienced regex user and least of all in Java's flavor. I had never heard the phrase 'matcher group' before so that error message meant nothing to me. 
I finally found  a good explanation of matcher/capturing/non-capturing groups in Java regex here: http://stackoverflow.com/questions/16517689/confused-about-matcher-group-in-java-regex#16582083 Worth cribbing for the Graylog docs? They are quite sparse in that area and could use more detail.
